org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/NullAnnotationTests18.java

```diff
@@ -628,4 +629,32 @@ public void testNullableVar() {
 				"----------\n";
 		runner.runNegativeTest();
 	}
+	public void testGH629() {
```

Comment:
I guess we have tests to make sure this kind of resolutions are working when null annotation are not enabled. May be something to think about to see if such test is needed.